plugins/decklink/decklink-output.cpp

```diff
@@ -0,0 +1,237 @@
+#include <obs-module.h>
+#include <obs-avc.h>
+
+#include "const.h"
+
+#include "DecklinkOutput.hpp"
+#include "decklink-device.hpp"
+#include "decklink-device-discovery.hpp"
+#include "decklink-devices.hpp"
+
+#include "../../libobs/media-io/video-scaler.h"
+
+static void decklink_output_destroy(void *data)
+{
+	auto *decklink = (DeckLinkOutput *)data;
+	delete decklink;
+}
+
+static void *decklink_output_create(obs_data_t *settings, obs_output_t *output)
+{
+	auto *decklinkOutput = new DeckLinkOutput(output, deviceEnum);
+
+	decklinkOutput->deviceHash = obs_data_get_string(settings, DEVICE_HASH);
+	decklinkOutput->modeID = obs_data_get_int(settings, MODE_ID);
+
+	return decklinkOutput;
+}
+
+static void decklink_output_update(void *data, obs_data_t *settings)
+{
+	auto *decklink = (DeckLinkOutput *)data;
+
+	decklink->deviceHash = obs_data_get_string(settings, DEVICE_HASH);
+	decklink->modeID = obs_data_get_int(settings, MODE_ID);
+}
+
+static bool decklink_output_start(void *data)
+{
+	auto *decklink = (DeckLinkOutput *)data;
+	struct obs_audio_info aoi;
+
+	if (!obs_get_audio_info(&aoi)) {
+		blog(LOG_WARNING, "No active audio");
+		return false;
+	}
+
+	decklink->audio_samplerate = aoi.samples_per_sec;
+	decklink->audio_planes = 2;
+	decklink->audio_size = get_audio_size(AUDIO_FORMAT_16BIT, aoi.speakers, 1);
+
+	decklink->start_timestamp = 0;
+
+	ComPtr<DeckLinkDevice> device;
+
+	device.Set(deviceEnum->FindByHash(decklink->deviceHash));
+
+	DeckLinkDeviceMode *mode = device->FindOutputMode(decklink->modeID);
+
+	decklink->SetSize(mode->GetWidth(), mode->GetHeight());
+
+	struct video_scale_info to = {};
+	to.format = VIDEO_FORMAT_UYVY;
+	to.width = mode->GetWidth();
+	to.height =  mode->GetHeight();
+
+	obs_output_set_video_conversion(decklink->GetOutput(), &to);
+
+	decklink->Activate(device, decklink->modeID);
+
+	audio_convert_info conversion{};
```

Comment:
Is this a struct initialization?  I'm having trouble finding an example of this syntax.